internal/metrics/otelmetrics/factory.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics
+
+import (
+	"context"
+	"log"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	metric "go.opentelemetry.io/otel/metric"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+)
+
+type otelFactory struct {
+	meter metric.Meter
+}
+
+func NewFactory(meterProvider *sdkmetric.MeterProvider) metrics.Factory {
+	otel.SetMeterProvider(meterProvider)
+	return &otelFactory{
+		meter: otel.Meter("jaeger-V2"),
```

Comment:
You are not even using passed provider.

Reply:
I am using it to set the provider as global provider, which can be called as otel.Meter.
Following this: 
![Screenshot 2024-06-24 202239](https://github.com/jaegertracing/jaeger/assets/103821431/c8ca1975-8002-4603-af31-5eb2ea504382)

Reply:
There is no need to depend on any globals

Reply:
Done